test/tests/kernels/vector_fe/tests

```diff
@@ -0,0 +1,8 @@
+[Tests]
+  [./vector_fe]
+    type = Exodiff
+    input = 'vector_kernel.i'
+    exodiff = 'vector_kernel_out.e'
+    abs_zero = 1e-9
+  [../]
```

Comment:
One test rules them all? :-)

Reply:
I've added a couple more tests. One demonstrates coupling a vector variable into a standard kernel. The other tests the Jacobian for the former case since we introduce a `_vector_phi` member on top of `_phi`.

I would like to add a LAGRANGE_VEC test, but without too much more work it will have to use integrated boundary conditions. It's currently possible in libmesh to specify dirichlet boundary conditions for LAGRANGE_VEC, but we'll have to add some additional machinery to MOOSE to have it work for us.